client/src/common/CurrentUserProvider.tsx

```diff
@@ -26,11 +28,21 @@ interface CurrentUserProviderProps {
 export function CurrentUserProvider({ children }: CurrentUserProviderProps) {
   const result = useQuery(GET_CURRENT_USER);
 
-  // If the current user is null, we need to redirect to SSO login
+  // If the current user is null, redirect to SSO login
   if (!result.loading && !result.data.currentUser) {
     window.location.replace(loginUrl);
   }
 
+  // If the user exists but setupComplete is false,
+  // redirect to them to the signup form
+  if (
+    result.data?.currentUser &&
+    !result.data.currentUser.setupComplete &&
+    !window.location.href.includes("/signup")
+  ) {
+    window.location.replace("/signup");
```

Comment:
this might break the web browsers back button

Reply:
I think it's usually preferred to use `replace` for redirects. Since it wasn't the user doing the navigation, it shouldn't go onto the history stack. I can change this to `assign` tho if you would prefer.